src/domain/community/user/user.service.authorization.ts

```diff
@@ -35,9 +35,9 @@ export class UserAuthorizationService {
       user.authorization
     );
 
-    user.authorization = this.appendCredentialRules(
+    user.authorization = await this.appendCredentialRules(
```

Comment:
This makes the method inconsistent with other implementations, where it is synchronous, e.g. organisation.service.authorization.ts. Either make it synchronous everywhere or asynchronous everywhere. Otherwise this inconsistency will bring problems.